apps/managedidentity/managedidentity_test.go

```diff
@@ -50,129 +61,182 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
+	responseBody := ErrorResponse{
 		Err:  err,
 		Desc: desc,
 	}
 	jsonResponse, e := json.Marshal(responseBody)
 	return jsonResponse, e
 }
 
-type resourceTestData struct {
-	source   Source
-	endpoint string
-	resource string
-	miType   ID
+func createMockFile(t *testing.T, path string, size int64) {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		t.Fatalf("failed to create directory: %v", err)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		t.Fatalf("failed to create file: %v", err)
+	}
+	defer f.Close() // Ensure the file is closed
+
+	if size > 0 {
+		if err := f.Truncate(size); err != nil {
+			t.Fatalf("failed to truncate file: %v", err)
+		}
+	}
+
+	// Write the content to the file
+	if _, err := f.WriteString("secret file data"); err != nil {
+		t.Fatalf("failed to write to file: %v", err)
+	}
+	t.Cleanup(func() { os.Remove(path) })
 }
 
-type errorTestData struct {
-	code          int
-	err           string
-	desc          string
-	correlationid string
+func getMockFilePath(t *testing.T) string {
+	tempDir := t.TempDir()
+	return filepath.Join(tempDir, "AzureConnectedMachineAgent")
 }
 
-func Test_SystemAssigned_Returns_AcquireToken_Failure(t *testing.T) {
-	testCases := []errorTestData{
-		{code: http.StatusNotFound,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusNotImplemented,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusServiceUnavailable,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusBadRequest,
-			err:           "invalid_request",
-			desc:          "Identity not found",
-			correlationid: "121212",
-		},
+func setEnvVars(t *testing.T, source Source) {
+	switch source {
+	case AzureArc:
+		t.Setenv(identityEndpointEnvVar, "http://127.0.0.1:40342/metadata/identity/oauth2/token")
+		t.Setenv(imdsEndVar, "http://169.254.169.254/metadata/identity/oauth2/token")
+	case AppService:
+		t.Setenv(identityEndpointEnvVar, "http://127.0.0.1:41564/msi/token")
+		t.Setenv(identityHeaderEnvVar, "secret")
+	case CloudShell:
+		t.Setenv(msiEndpointEnvVar, "http://localhost:40342/metadata/identity/oauth2/token")
+	case ServiceFabric:
+		t.Setenv(identityEndpointEnvVar, "http://localhost:40342/metadata/identity/oauth2/token")
+		t.Setenv(identityHeaderEnvVar, "secret")
+		t.Setenv(identityServerThumbprintEnvVar, "thumbprint")
 	}
+}
 
-	for _, testCase := range testCases {
-		t.Run(http.StatusText(testCase.code), func(t *testing.T) {
-			fakeErrorClient := mock.Client{}
-			responseBody, err := makeResponseWithErrorData(testCase.err, testCase.desc)
-			if err != nil {
-				t.Fatalf("error while forming json response : %s", err.Error())
-			}
-			fakeErrorClient.AppendResponse(mock.WithHTTPStatusCode(testCase.code),
-				mock.WithBody(responseBody))
-			client, err := New(SystemAssigned(), WithHTTPClient(&fakeErrorClient))
+func unsetEnvVars(t *testing.T) {
```

Comment:
If all tests that set environment variables do so via `t.Setenv` (looks like they do), you don't need this function